### ISSUE TITLE
Chromebook Usage Prevents Linux Traffic

### ISSUE THREAD

**Issue by mccabeservant** [krlvm/PowerTunnel-Android#39]
When I install this software on my Lenovo AMD64 Chromebook, which basically runs three different OSs, I have issues with Crostini Linux. The following describes what happens with each OS...

1. Apps running in the Android environment (like a version of Firefox downloaded from the Google Play store) work well.
2. The native Chrome browser in the ChromeOS itself works well.
3. Apps running in the Linux environment (like a DEB version of Firefox) can no longer access any websites at all.

Any thoughts on why this would be the case, or suggestions on how I could adjust the code so that it will allow all web traffic to pass through the VPN and proxy on Chromebook, regardless of the OS?

**Comment by krlvm** [comment]
Have you tried to run the [Linux version](https://github.com/krlvm/PowerTunnel) of PowerTunnel in the Linux environment and direct all traffic across the system to it?

**Comment by mccabeservant** [comment]
Is there an easy way to tell, using the compiled JAR executable, if traffic is actually going through the VPN and proxy?

I tried turning on the Journal option, then restarted the device, started PowerTunnel, visited some sites in Firefox, and opened Journal, but all Journal showed me was an empty white box. I tried visiting sites in Firefox with the Journal window open, but still nothing showed up. I tried this both on my Chromebook and also on an Ubuntu VM on my Mac. Is Journal supposed to show me my own traffic?

If not, is there another way for me to verify that traffic is actually going through it?

With the Android VPN, I could go into the Settings app and see the VPN turned on there. Plus, on Chromebook, it blocked my Linux traffic when it was on, so I was pretty sure traffic was actually hitting it. But the Linux app doesn't create a VPN in Chromebook's Settings. I don't know if it is supposed to...?

If there is no easy way to tell, I can try to go through the code and create some way to tell -- write it to a log file or something -- then recompile and try again. But if something that will tell me already exists, that would be much simpler for me!

(As a side note, the Options screen is so tall that the Save button was off my small Chromebook screen and getting to it was extremely difficult.)

**Comment by mccabeservant** [comment]
I've gotten the Journal option to work now on Ubuntu, and I see all my traffic going through it (as well as through the terminal that I used to launch PowerTunnel), so I understand what kinds of things need to be done to get it to work on Crostini. I'll see if I can make that happen.

**Comment by krlvm** [comment]
The Linux app does not create a VPN profile, it's a proxy server, which you should setup manually. This proxy server also is running under the VPN connection established by the Android version, so you can run proxy from the Android version, without installing the Desktop one.

If you can get it working, you have a two ways:

1. Exclude Linux container app from Android version settings and setup your Linux apps to use 127.0.0.1:8085 proxy server. You need to adjust IP address if localhost is not accessible from the Linux container.
2. Run PowerTunnel Desktop (Linux) version in Linux environment and setup all the apps in all environments to use it. You need to adjust IP address if Linux container's localhost is not accessible outside.

**Comment by mccabeservant** [comment]
Thank you for the information!

**Comment by krlvm** [comment]
Have you resolved the issue?

**Comment by mccabeservant** [comment]
My issue is not exactly resolved, but it isn't a bug with your software. Your Android software works for Android, even Chromebook Android; and your Linux software sets up a proxy server like it is supposed to. So I see no need to ask you to fix anything.

What's more, you've given me helpful information that I can use to find a solution that will ultimately work for me. I have much more research to do, but I am not expecting anything more from you.

I was able to get the Linux proxy server code running on Crostini (Chromebook-Linux) and point Firefox to it. I have not yet been able to get other Linux web apps (specifically, Chromium) to use that proxy server though. But what I was really looking for was something that will automatically make all apps on a Chromebook use a particular proxy without any real manual configuration. Your apps are the closest thing I have found so far!

Thank you!
